support-frontend/test/selenium/contributions/pages/ContributionsLanding.scala

```diff
@@ -8,7 +8,7 @@ case class ContributionsLanding(region: String, testUser: TestUser)(implicit val
 
   val url = s"${Config.supportFrontendUrl}/$region/contribute"
 
-  private val contributeButton = className("form__submit-button")
+  private val contributeButton = cssSelector("#contributions-landing-submit-contribution-button")
```

Comment:
by class name with an additional class name was fine to be honest, the main thing was the clue in the name that it realtes to the test.  Even now, the actual name itself doesn't make it clear it's for the tests (making it harder to review the test for compliance with the new "rule".
I realise the fashion is against overloading names with extra type information, but this area is pretty weakly typed.

Reply:
I think we should use ID (or another attr) to keep our separation of concerns clear. I think it's best if we keep class to indicate the way that something will be referenced on the DOM to apply styles. By keeping it separate, we are indicating this is an attribute which can't be arbitrarily changed and only checked against styles

Reply:
fair enough originally I just thought class might be simpler :+1:   I am a bit stronger on the idea of putting a prefix to make it more obvious that it's an ID designed for this purpose rather than any other.
But either way better to get this shipped and then address the niceties in a followup to avoid having broken tests for longer than necessary

Reply:
Cool i've added the qa prefix - now will merge when team city gives a passing build 👍